pymc3/tests/test_transforms.py

```diff
@@ -274,11 +274,11 @@ def test_chain_jacob_det():
 
 
 class TestElementWiseLogp(SeededTest):
-    def build_model(self, distfam, params, size, transform, testval=None):
+    def build_model(self, distfam, params, *, size=None, shape=None, transform=None, testval=None):
         if testval is not None:
             testval = pm.floatX(testval)
         with pm.Model() as m:
-            distfam("x", size=size, transform=transform, testval=testval, **params)
+            distfam("x", size=size, shape=shape, transform=transform, testval=testval, **params)
```

Comment:
All of the changes in this module need to be reverted; they have nothing to do with the `shape` or `dims` feature, nor should they.

Reply:
All of the changes in this module are because many of these tests are actually parametrized by `shape`.
Consider this test case from before the change:
```python
    @pytest.mark.parametrize(
        "a,b,size",
        [
            (1.0, 1.0, (2,)),                               # scalar is implied; 2 replicate dimensions
            (np.ones(3), np.ones(3), (4, 3)),               # (3,) is implied; (4, 3) replicate dimensions
        ],
    )
    def test_beta_ordered(self, a, b, size):
        testval = np.sort(np.abs(np.random.rand(*size)))    # but the testval is at most 2-dimensional
        model = self.build_model(
            pm.Beta,
            {"alpha": a, "beta": b},
            size=size,                     # in the second parametrize case the RV is expanded to (4, 3, 3)
            testval=testval,               # but the testval is only (4, 3)
            transform=tr.Chain([tr.logodds, tr.ordered]),
        )
        self.check_vectortransform_elementwise_logp(model, vect_opt=0)
```

Comparing with `master` we can see that the test was up to 2-dimensional to begin with. The change to `size` had changed the behaviour of the test - I'm just reverting it to what was originally intended & tested.
https://github.com/pymc-devs/pymc3/blob/2dee9dafb3541043ae23b66fe9262e70fbd2769a/pymc3/tests/test_transforms.py#L405-L421

Yes, the test could be refactored to use `size`, by either replicating the `shape`←→`size` logic, or parametrizing the test with both `shape` and `size` manually. That would be a lot of unnecessary work and make the tests harder to work with!
Since this PR is about restoring `shape` as a backwards-compatible and fully supported way to parametrize RVs I do not see the point in making our lives harder by notoriously avoiding it in tests like this one.

Reply:
I checked the `testval`s for this test in `v3` and `v4` and they're both the same.  The symbolic types for the resulting variables are also the same between `v3` and `v4`, so I don't understand what's changed here&mdash;aside from the switch to `size`.

More importantly, this is a scalar variable, so `size` and `shape` should be the same.  If they're not, then that's an altogether different issue that needs to be addressed separately.

Regardless, the change here is still an unnecessary reversion that forces these tests to use an irrelevant and newly added feature.